### PR TITLE
Add navbar

### PR DESCRIPTION
**Pull request by upacyxou**
- bwl logo typography added
- useThrottle hook added
- navbar added

### CODE REVIEW

File: src/components/Button.tsx [hunk 1]
```diff
@@ -56,17 +56,27 @@ interface ButtonProps {
...
 
 export default function ({
   outlined,
   children,
   small,
   icon,
+  url,
   ...rest
-}: React.HTMLAttributes<HTMLButtonElement> & ButtonProps) {
+}: Omit<React.HTMLAttributes<HTMLButtonElement>, 'icon'> & ButtonProps) {
```
Comment (T-Damer) [inline src/components/Button.tsx:69]: Can we get rid of `React` here?

File: src/components/Button.tsx [hunk 2]
```diff
@@ -56,17 +56,27 @@ interface ButtonProps {
...
   return (
-    <button className={button({ outlined, small, icon })} {...rest}>
+    <button
+      className={button({ outlined, small, icon })}
+      onClick={() => {
```
Comment (T-Damer) [inline src/components/Button.tsx:73]: Get rid of curly braces here

File: src/components/Button.tsx [hunk 3]
```diff
@@ -56,17 +56,27 @@ interface ButtonProps {
...
+        if (url) {
```
Comment (T-Damer) [inline src/components/Button.tsx:74]: And here

File: src/components/Logo.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  justifyContent('justify-center'),
+  space('space-y-2', 'tiny:space-y-0', 'tiny:space-x-2'),
+  whitespace('whitespace-pre-line')
+)
+export default function ({ large }: { large?: boolean }) {
+  const { xs } = useBreakpoints()
+
+  return (
+    <span className={container}>
+      {large || (!large && !xs) ? (
```
Comment (T-Damer) [inline src/components/Logo.tsx:26]: You can do `&&`, not `? :` probably

File: src/components/LogoFooter.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  justifyContent('justify-center'),
+  alignItems('items-center'),
+  space('space-y-10')
+)
+const linksContainer = classnames(
+  display('flex'),
+  flexDirection('flex-row'),
+  space('space-x-8')
+)
+export default function () {
```
Comment (T-Damer) [inline src/components/LogoFooter.tsx:24]: Add blank line between classnames and export

File: src/components/LogoFooter.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import Twitter from 'icons/Twitter'
+import classnames, {
+  alignItems,
+  display,
+  flexDirection,
+  justifyContent,
+  space,
+} from 'classnames/tailwind'
+
+const container = classnames(
```
Comment (T-Damer) [inline src/components/LogoFooter.tsx:12]: Maybe let's name it `wrapper` or `box`? You already have a container here

File: src/components/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+const buttonsContainer = classnames(
+  display('flex'),
+  alignItems('items-center'),
+  space('space-x-4', 'lg:space-x-10')
+)
+
+export default function () {
+  const { md } = useBreakpoints()
+
+  const [isOpen, setIsOpen] = useState(false)
```
Comment (T-Damer) [inline src/components/Navbar.tsx:46]: Good solution, if you want - you can go with pure `tailwind`, like I did for `dosu`: https://github.com/BigWhaleLabs/dosu-frontend/blob/main/src/components/Navbar.tsx

File: src/components/Navbar.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  return (
+    <nav ref={navbarRef} className={navbar(backgroundVisible)}>
+      <Logo />
+      <div className={buttonsContainer}>
+        {md && <NavbarLinks />}
+
+        <Button
+          outlined
+          small={!md}
+          onClick={() => window.open('https://discord.gg/UtFAnyATNR', '_blank')}
```
Comment (T-Damer) [inline src/components/Navbar.tsx:70]: You have a `url` prop which will do the same here, isn't it?

File: src/components/Root.tsx [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import { classnames, margin, padding } from 'classnames/tailwind'
+import ChildrenProp from 'models/ChildrenProp'
+
+const root = classnames(margin('mx-auto'), padding('pb-10'))
```
Comment (T-Damer) [inline src/components/Root.tsx:4]: `pb-10` is for `Footer`? I think we shouldn't do this, maybe we can do something inside the `Footer` itself?

File: src/components/Navbar.tsx (hunk 7 above)
Comment (T-Damer) [reply src/components/Navbar.tsx:46]: Of course, you'll need to change the `transform` and `inset` properties to make the menu be above, not on the right side

**Review by T-Damer — CHANGES_REQUESTED** [review]
Gave a few cosmetic comments 👍

File: src/components/Button.tsx (hunk 1 above)
Comment (smyzin) [reply src/components/Button.tsx:69]: Yep, removed it

File: src/components/Button.tsx (hunk 2 above)
Comment (smyzin) [reply src/components/Button.tsx:73]: I think it's better to leave them as they are. It's a standard syntax

File: src/components/Button.tsx (hunk 3 above)
Comment (smyzin) [reply src/components/Button.tsx:74]: Removed

File: src/components/Logo.tsx (hunk 4 above)
Comment (smyzin) [reply src/components/Logo.tsx:26]: I tried but failed in this. Compiler replacing with `large || (!large && !xs && (<></>))` and breaking the logic. Left it as it is for now

File: src/components/LogoFooter.tsx (hunk 6 above)
Comment (smyzin) [reply src/components/LogoFooter.tsx:12]: Did it. Thx

File: src/components/Navbar.tsx (hunk 8 above)
Comment (smyzin) [reply src/components/Navbar.tsx:70]: You're right. Changed it. Thanks

File: src/components/Root.tsx (hunk 9 above)
Comment (smyzin) [reply src/components/Root.tsx:4]: Why not? It's a root wrapper, and Footer is `absolute`. Small padding at the end of the page is not a bad idea as I think

File: src/components/Logo.tsx (hunk 4 above)
Comment (T-Damer) [inline src/components/Logo.tsx:26]: Does this mean that you can do `large || !xs`?

File: src/components/Navbar.tsx (hunk 7 above)
Comment (T-Damer) [reply src/components/Navbar.tsx:46]: Let's have both approaches and take a look, which one is better, for dosu or for bwl

File: src/components/Logo.tsx (hunk 4 above)
Comment (smyzin) [reply src/components/Logo.tsx:26]: Nope. This component is used in two other components. In the `navbar` & `LogoFooter`. One is used with `large` and the other without it. A need to hide text only in the navbar, when the screen is xs size

File: src/components/Root.tsx (hunk 9 above)
Comment (T-Damer) [reply src/components/Root.tsx:4]: Let's update the `Footer` then:
```
.grid-wrapper {
  width: 100vw;
  height: 50vh;
  position: relative;
  perspective: 180px;
  perspective-origin: 50% 20%;
  overflow: hidden;
}

```

Comment (smyzin) [reply src/components/Root.tsx]: Changed it. Everything works fine. Thank you

File: src/components/Root.tsx [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { classnames, margin } from 'classnames/tailwind'
+import ChildrenProp from 'models/ChildrenProp'
+
+const root = classnames(margin('mx-auto'))
+
+export default function ({ children }: ChildrenProp) {
```
Comment (T-Damer) [inline src/components/Root.tsx:6]: Actually, it't not a `Root`, it's `InnerRoot` or `InnerWrapper`, something like that

Comment (T-Damer) [reply src/components/Root.tsx:6]: Maybe it's better to have such styles inside the `index.css` > `body` 🤔

Comment (T-Damer) [reply src/components/Root.tsx:6]: Nah, this won't work

**Pull request merged**